editors/code/src/cargo.ts

```diff
@@ -43,30 +75,15 @@ export class Cargo {
             throw new Error(`Cargo invocation has failed: ${err}`);
         }
 
+        if (spec.filter) {
+            artifacts = spec.filter(artifacts);
+        }
+
         return artifacts;
```

Comment:
Looks like we can avoid the mutable variable here
```suggestion
      return spec?.filter(artifacts) ?? [];

```

Reply:
Perhaps
```ts
return spec.filter?.(artifacts) ?? artifacts;
```
if the filter is null we still need artifacts